lib/tpm2.c

```diff
@@ -1073,14 +1079,16 @@ uint32_t fix_esys_hierarchy(TPMI_RH_HIERARCHY hierarchy)
             return ESYS_TR_RH_ENDORSEMENT;
         case TPM2_RH_PLATFORM:
             return ESYS_TR_RH_PLATFORM;
+        case TPM2_RH_PLATFORM_NV:
+            return ESYS_TR_RH_PLATFORM_NV;
         default:
-            return TSS2_ESYS_RC_BAD_VALUE;
+            LOG_ERR("An unknown hierarchy handle was passed: 0x%08x", hierarchy);
+            return 0xffffffff;
```

Comment:
This will cause something else to fail right? Is this an exceptional case, can someone actually reach this?

Reply:
Should not be reachable, but was reached before this PR. So this is more for developers than users.
I could make this an `assert(0)` if you prefer.

Reply:
Id rather just make this an error and have the caller handle it